packages/design-system/src/components/TextField/useLabelMask.tsx

```diff
@@ -93,37 +93,58 @@ export const SSN_MASK: MaskFunction = makeMask(RE_SSN, '###-##-####', (match) =>
  * Currency mask is a little different, we need to modify the incoming content to strip
  * out any commas or dollar signs before evaluating it via the Intl.NumberFormat function.
  */
-export const CURRENCY_MASK = (rawInput = '', valueOnly = false) => {
-  const signed = rawInput.includes('-');
-  const stripped = rawInput.replace(/[^0-9.]/g, '');
+export const CURRENCY_MASK = makeMask(RE_CURRENCY, '$', (match) => {
+  const signed = match[0].includes('-');
+  const stripped = match[0].replace(/[^0-9.]/g, '');
   const clipped = stripped.includes('.') ? stripped.slice(0, stripped.indexOf('.') + 3) : stripped;
   const USDollar = new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' });
-  const formatted = USDollar.format(Number(clipped));
+  const formatted = USDollar.format(Number(clipped)).replace(/\.00/, '');
```

Comment:
These names are funny 😆. So succinct and descriptive and rhythmic. I feel like it's part of a song until I get to `USDollar`. You know...the US Dollar formatter could go outside the function so we don't break up the song 🎼

Reply:
Lol :) it's like 'signed, sealed delivered' by Stevie Wonder... 'Signed, stripped, form-matted 🎵'